napari/components/_dims/model.py

```diff
@@ -1,143 +1,255 @@
-
-        curr_dims = len(self.indices)
-
-        if curr_dims > max_dims:
-            self.indices = self.indices[:max_dims]
-            dims = curr_dims
-        else:
-            dims = max_dims
-            self.indices.extend([0] * (max_dims - curr_dims))
-
-        for dim in range(2, dims):  # do not create sliders for y/x-axes
-            try:
-                dim_len = max_shape[dim]
-            except IndexError:
-                dim_len = 0
-            self.events.update_slider(dim=dim, dim_len=dim_len,
-                                      max_dims=max_dims)
-
-    def _slider_value_changed(self, value, axis):
-        self.indices[axis] = value
-        self._need_redraw = True
-        self._update()
-
-    def _calc_max_dims(self):
-        """Calculates the number of maximum dimensions in the contained images.
-        """
-        max_dims = 0
+        slice_list = []
+        project_list = []
+        z = zip(self.mode, self.display, self.point, self.interval, self.range)
+        for (mode, display, point, interval, range) in z:
+            if mode == DimsMode.POINT or mode is None:
+                if display:
+                    # no slicing, cropping or projection:
+                    project_list.append(False)
+                    slice_list.append(slice(None, None, None))
+                else:
+                    # slice:
+                    project_list.append(False)
+                    slice_list.append(int(round(point)))
+            elif mode == DimsMode.INTERVAL:
+                if display:
+                    # crop for display:
+                    project_list.append(False)
+                    if interval is None:
+                        slice_list.append(slice(None))
+                    else:
+                        slice_list.append(slice(int(round(interval[0])),
+                                          int(round(interval[1]))))
+                else:
+                    # crop before project:
+                    project_list.append(True)
+                    if interval is None:
+                        slice_list.append(slice(None))
+                    else:
+                        slice_list.append(slice(int(round(interval[0])),
+                                          int(round(interval[1]))))
+
+        slice_tuple = tuple(slice_list)
+        project_tuple = tuple(project_list)
+
+        return slice_tuple, project_tuple
 
-        for layer in self.viewer.layers:
-            dims = layer.ndim
-            if dims > max_dims:
-                max_dims = dims
+    @property
+    def indices(self):
+        """
+        Indices for slicing
 
-        self._max_dims = max_dims
+        Returns
+        -------
+        slice : tuple
+            The slice tuple
+        """
+        return self.slice_and_project[0]
 
-        self._need_slider_update = True
-        self._update()
+    def set_all_ranges(self, all_ranges: Sequence[Union[int, float]]):
```

Comment:
why are we using a setter method instead of just making `range` a property?

Reply:
i have made this a property